### PR TITLE
React Wrapper

### LINKED ISSUE #127: Web3API Client React Wrapper
Create a react wrapper around the Web3API JS Client, which enables to easily initialize the client & send queries to Web3APIs throughout their application.

We'd want to support both a Component & Hook interface. The Web3API Client will be created using a React provider, and passed through to all queries as a context.

A possible developer experience might look like:
```jsx
// Component Interface
<Web3ApiClient redirects={...}>
  <MyApp />
</Web3ApiClient>

// elsewhere in MyApp
<Web3ApiQuery uri={...} query={...} variables={...}>
{(result: QueryResult) => {
  // Do something with the result
}}
</Web3ApiQuery>
```
```jsx
// Hook Interface
// Web3ApiClient as we did above...

const { loading, error, data } = useWeb3ApiQuery({ uri, query, variables });
```

### PR DESCRIPTION
**Pull request by cbrzn**
closes: #127

### CODE REVIEW

File: packages/js/client/src/declaration.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+declare module "spawn-command";
```
Comment (mpetrunic) [inline packages/js/client/src/declaration.d.ts]: It would probably make sense to also declare method interface

File: packages/js/react/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "@testing-library/react": "^11.2.5",
+    "@testing-library/react-hooks": "^5.1.0",
+    "@types/jest": "26.0.8",
+    "jest": "26.2.2",
+    "jest-environment-jsdom": "^26.0.1",
+    "rimraf": "3.0.2",
+    "typescript": "^4.1.3"
+  },
+  "peerDependencies": {
+    "@react": ">=16.13.1"
```
Comment (mpetrunic) [inline packages/js/react/package.json]: maybe we can relax this to accept all 16.x versions?

File: packages/js/client/src/declaration.d.ts (hunk 1 above)
Comment (cbrzn) [reply packages/js/client/src/declaration.d.ts]: Okok agreed

File: packages/js/react/package.json (hunk 2 above)
Comment (cbrzn) [reply packages/js/react/package.json]: Sure :-D

**Review by morrigan — DISMISSED** [review]
Looks great!

File: demos/simple-storage/dapp/package.json [hunk 3]
```diff
@@ -35,6 +33,10 @@
     "prestart": "yarn install:web3api",
     "install:web3api": "npx web3api-worker-install ./public"
   },
+  "resolutions": {
+    "@types/react": "^16.9.0",
+    "@types/react-dom": "^16.9.0"
+  },
```
Comment (dOrgJelli) [inline demos/simple-storage/dapp/package.json:39]: Just curious, why is this needed?

File: packages/js/react/src/__tests__/env/setup.js [hunk 4]
```diff
@@ -0,0 +1 @@
+global.XMLHttpRequest = undefined;
```
Comment (dOrgJelli) [inline packages/js/react/src/__tests__/env/setup.js:1]: Why is this needed?

File: packages/js/react/src/__tests__/env/index.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+/**
+ * A custom environment to set the TextEncoder that is required by ../client/src/wasm/WasmWeb3Api.ts.
+ */
+module.exports = class CustomTestEnvironment extends Environment {
+  async setup() {
+    await super.setup();
+    if (typeof this.global.TextEncoder === "undefined") {
+      const { TextEncoder } = require("util");
+      this.global.TextEncoder = TextEncoder;
```
Comment (dOrgJelli) [inline packages/js/react/src/__tests__/env/index.ts:11]: Why is this needed?

**Review by dOrgJelli — DISMISSED** [review]

File: demos/simple-storage/dapp/package.json (hunk 3 above)
Comment (cbrzn) [reply demos/simple-storage/dapp/package.json:39]: IIRC: since we are doing this is a monorepo, when we added the react dependencies in the react package, it wasn't working because it was taking the versioning from this project (dapp) so we had to create this resolutions attribute which tells yarn to use this versioning in this project

File: packages/js/react/src/__tests__/env/setup.js (hunk 4 above)
Comment (cbrzn) [reply packages/js/react/src/__tests__/env/setup.js:1]: We were having a cors error in the tests, this fixed it

File: packages/js/react/src/__tests__/env/index.ts (hunk 5 above)
Comment (cbrzn) [reply packages/js/react/src/__tests__/env/index.ts:11]: It was saying that TextEncoder was undefined in the tests, this fixed it

File: demos/simple-storage/dapp/package.json (hunk 3 above)
Comment (dOrgJelli) [reply demos/simple-storage/dapp/package.json:39]: Ah okay, makes sense

File: packages/js/react/src/__tests__/env/index.ts (hunk 5 above)
Comment (dOrgJelli) [reply packages/js/react/src/__tests__/env/index.ts:11]: Ah okay sounds good. Probably just a jest browser testing issue. Nothing to worry about in production, as the demo dApp works just fine.

**Review by dOrgJelli — APPROVED** [review]
